### PR TITLE
Allow wrap/unwrap toggle across elements

### PR DESCRIPTION
**Pull request by mightyiam**
Initial report suggests an issue. Not sure whether or how much this was possible in previous versions.

### CODE REVIEW

File: src/test/unit/specific_feature_tests/exec_toggles.js [hunk 1]
```diff
@@ -318,11 +318,11 @@ test("IE unwraps partially subscript selection", function () {
     );
 });
 
-function doesntWrapAcrossRootContainers(command) {
-    var noChangeHtml = "<p>Foo</p><p>Bar</p>";
+function wrapsAcrossRootContainers(command) {
+    var tagName = getBrowserTagname(command);
     manipulationTestHelper({
-        startHtml: noChangeHtml,
-        prepareFunf: function (wymeditor) {
```
Comment (mightyiam) [inline src/test/unit/specific_feature_tests/exec_toggles.js]: Had a typo here previously! Bad test! How do I avoid this in JavaScript?

Comment (kylegibson) [inline src/test/unit/specific_feature_tests/exec_toggles.js]: `manipulationTestHelper` could do a validation on the object it receives, and if there are any keys missing, or unexpected keys passed it, it aborts.

**Comment by kylegibson** [comment]
The code looks good. However, I notice that the implementation and tests focus on "root-level" containers, when the bug report just mentions items, presumably non-root level containers.

1. Does this fix work with non-root level containers?
2. Are there any tests for this on non-root level containers?
3. The previous implementation did not allow wrap/unwrap on root level containers. What was the rationale for that previous behavior, and why are we changing it now? (I could see case where non-root level containers would be wrap/unwrap, but not root level)

**Comment by winhamwr** [comment]
Tracing this back to #632 makes me feel more comfortable with this fix.

**Comment by winhamwr** [comment]
I think the answer to Kyle's questions are:

1. I think yes, because the [selectedContainer()](https://github.com/wymeditor/wymeditor/blob/master/src/wymeditor/editor/base.js#L849) check was disallowing the wrap acorss any sibling-level selection (or anything other than a selection from a parent to a child). 
2. No. I think at least one test across e.g. sibling items in a sub-list would be useful.
3. I looked through #632 and I can't figure out for the life of me why we thought it would be ideal to not allow people to bold across root level containers. Do you remember @mightyiam?

**Comment by mightyiam** [comment]
1. What Wes wrote.
2. I'll make such a test.
3. It is not easy to understand the implications of changes in such a complex UI (most of which is whatever the current browser decides to do).
  And I'm not really using the editor for anything so, I think I need to eat my own :dog2: food.

File: src/test/unit/specific_feature_tests/exec_toggles.js [hunk 2]
```diff
@@ -338,67 +338,187 @@ function doesntWrapAcrossRootContainers(command) {
...
-test("Doesn't unwrap superscript across root containers", function () {
-    doesntUnwrapAcrossRootContainers(WYMeditor.EXEC_COMMANDS.SUPERSCRIPT);
+test("unwraps superscript from inside `li` to child of sibling `li`", function () {
+    unwrapsFromInsideLiToChildOfSiblingLi(WYMeditor.EXEC_COMMANDS.SUPERSCRIPT);
 });
 
-test("Doesn't unwrap subscript across root containers", function () {
-    doesntUnwrapAcrossRootContainers(WYMeditor.EXEC_COMMANDS.SUPERSCRIPT);
+test("unwraps subscript from inside `li` to child of sibling `li`", function () {
+    unwrapsFromInsideLiToChildOfSiblingLi(WYMeditor.EXEC_COMMANDS.SUBSCRIPT);
```
Comment (mightyiam) [inline src/test/unit/specific_feature_tests/exec_toggles.js]: This is a mess. Perhaps in the commit, it is easier to see that I added some more tests.

**Comment by mightyiam** [comment]
New tests passed on supported browsers.

**Comment by winhamwr** [comment]
Other than the jshint problems, looks good!

**Comment by mightyiam** [comment]
Now I know why you came down to that multiline style, Wes.

**Comment by VINTproYKT** [comment]
Nicely done! I think it will be more convenient to apply some styles directly on paragraphs and other  blocks. This is exactly what will reveal (or improve) a concept of WYSIWYM. After I done with my basic model of app, I'll try to show what I mean.

**Pull request merged**